mlp/neuralnetwork.py

```diff
@@ -209,26 +228,41 @@ def empirical_error(self, y):
     def average_loss(self, y):
         """
         Compute the average loss function
-        :param y: numpy array of the expected classes
+        :param y: numpy array of the expected classes [5, 0, 4, ...] of size n
         :return:
         """
-        prediction = np.multiply(self._out, onehot_matrix(self.nb_out, y))
-        precision = np.max(prediction, axis=1)
-        log_err = np.multiply(np.log(precision), -1)
```

Comment:
* np.log_2(0) was causing problems when sampling from normal
* the rest i think i should revert back to how it was, because cross-entropy reduces to -log-likelihood in classification